gtf/middleware.py

```diff
@@ -32,16 +32,19 @@ def generic_template_finder_view(request, base_path='', extra_context={}):
     for t in possibilities:
         try:
             response = render(request, t, extra_context)
-        except (TemplateDoesNotExist):
+        except (TemplateDoesNotExist, NotADirectoryError):
             continue
         except OSError as e:
             # If there's a directory that matches the template we're looking for,
             # Django will raise a `IsADirectoryError` in `render` instead of a
             # `TemplateDoesNotExist` error. IsADirectoryError was introduced in
             # Python 3 and is a subclass of OSError and its errno corresponds to EISDIR,
-            # so for Python 2 compatibility, OSError is caught instead of IsADirectoryError
+            # but the error is mapped to the Exception NotADirectoryError.
+            # So for Python 2 compatibility, OSError is caught instead of IsADirectoryError
             if e.errno == errno.EISDIR:
                 continue
+            elif e.errno == errno.ENAMETOOLONG:
+                raise Http404('File name too long')
```

Comment:
We don't need to and shouldn't surface this information to website users. It's only useful to us in logging since it concerns how the software is failing. Either remove it, or use the logging module to log it.